NamelessSpigot/src/com/namelessmc/plugin/spigot/WhitelistRegistered.java

```diff
@@ -38,15 +40,21 @@ public void run() {
 				e.printStackTrace();
 				return;
 			}
+			
+			Config.MAIN.getConfig().getStringList("permanent-whitelisted")
+			.forEach(playername -> permWhitelistUUID.add(playername.getUniqueId()));		
 
 			Bukkit.getScheduler().runTask(NamelessPlugin.getInstance(), () -> {
 				for (final OfflinePlayer whitelistedPlayer : Bukkit.getWhitelistedPlayers()) {
 					if (!uuids.contains(whitelistedPlayer.getUniqueId())) {
-						// The player is whitelisted, but no(t) (longer) registered.
-						whitelistedPlayer.setWhitelisted(false);
-						uuids.remove(whitelistedPlayer.getUniqueId());
-						if (log) {
-							logger.info("Removed " + whitelistedPlayer.getName() + " from the whitelist.");
+						// The player is not on the permanent whitelist exclusion list
+						if (!permanentWhitelisted.contains(whitelistedPlayer.getUniqueId())) {						
+							// The player is whitelisted, but no(t) (longer) registered.
+							whitelistedPlayer.setWhitelisted(false);
+							uuids.remove(whitelistedPlayer.getUniqueId());
```

Comment:
It was part of the original code, not sure why. You can remove `uuids.remove(whitelistedPlayer.getUniqueId());`, it makes no sense because this code is in an `if (!uuids.contains` block